runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -2249,147 +2261,1096 @@ benchmarks! {
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::BuyNow(price),
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    cancel_buy_now {
+        let (
+            video_id,
+            (curator_account_id, actor, channel_id, _)
+        ) = setup_worst_case_scenario_mutable_video::<T>(
+            Some(T::MaxNumberOfAssetsPerVideo::get()),
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+        )?;
+
+        let (_, to_member) = member_funded_account::<T>(MEMBER_IDS[1]);
+        let price = BUY_NOW_PRICE.into();
+
+        let _ = setup_nft_in_buy_now::<T>(
+            curator_account_id.clone(),
+            actor,
+            video_id,
+            false,
+            price,
+        )?;
+
+        set_all_channel_paused_features::<T>(channel_id);
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+    }: _ (origin, actor, video_id)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::Idle,
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    update_buy_now_price {
+        let (
+            video_id,
+            (curator_account_id, actor, channel_id, _)
+        ) = setup_worst_case_scenario_mutable_video::<T>(
+            Some(T::MaxNumberOfAssetsPerVideo::get()),
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+        )?;
+
+        let price = BUY_NOW_PRICE.into();
+
+        let _ = setup_nft_in_buy_now::<T>(
+            curator_account_id.clone(),
+            actor,
+            video_id,
+            false,
+            price,
+        )?;
+
+        set_all_channel_paused_features::<T>(channel_id);
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let new_price = (BUY_NOW_PRICE + 1).into();
+    }: _ (origin, actor, video_id, new_price)
+        verify {
+            assert!(matches!(Pallet::<T>::video_by_id(video_id).nft_status, Some(Nft::<T> {
+                transactional_status: TransactionalStatus::<T>::BuyNow(new_price),
+                ..
+            })));
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
```

Comment:
State complexity comment still missing conditions like:
- curator group should have max number of permissions per level
- channel should have max number of collaborators
- complete payment has max complexity: (...)